orttraining/orttraining/python/training/ortmodule/experimental/json_config/_load_config_from_json.py

```diff
@@ -0,0 +1,240 @@
+    ortmodule_config_accessor._debug_options = debug_options
+
+def _define_load_function_keys():
+    """Define static key variables for each loading function"""
+
+    _load_propagate_cast_ops.loading_key = "PropagateCastOps"
+    _load_use_external_gpu_allocator.loading_key = "UseExternalGPUAllocator"
+    _load_enable_custom_autograd_function.loading_key = "EnableCustomAutogradFunction"
+    _load_allow_layer_norm_mod_precision.loading_key = "AllowLayerNormModPrecision"
+    _load_enable_grad_acc_optimization.loading_key = "EnableGradAccOptimization"
+    _load_run_symbolic_shape_infer.loading_key = "RunSymbolicShapeInference"
+    _load_use_static_shape.loading_key = "UseStaticShape"
+    _load_skip_check.loading_key = "SkipCheck"
+    _load_debug_options.loading_key = "DebugOptions"
+
+def load_from_json(ortmodule, path=None):
+    """Load config from json file at given path.
+
+    Here is the schema that the json file must adhere to:
+    {
+        "PropagateCastOps":
+        {
+            "Strategy": "FLOOD_FILL", # str representing strategy (like "NONE", "FLOOD_FILL"...)
+            "Level": 3, # propagate cast ops level as an int
+            "Allow": ["ABC", "DEF"] # propagate cast ops allow as list of strs
+        },
+        "UseExternalGPUAllocator" : false, # bool flag
+        "EnableCustomAutogradFunction": true, # bool flag
+        "AllowLayerNormModPrecision": true, # bool flag
+        "EnableGradAccOptimization": true, # bool flag
+        "UseStaticShape": true, # bool flag
+        "RunSymbolicShapeInference": false, # bool flag
+        "SkipCheck": # list of strs representing `_SkipCheck`s checks to skip which will be aggregated using |
+        [
+            "SKIP_CHECK_DEVICE",
+            "SKIP_CHECK_BUILD_GRADIENT",
+            "SKIP_CHECK_EXECUTION_AGENT"
+        ],
+        "DebugOptions": # debug options for user facing configuration
+        {
+            "LogLevel": "VERBOSE",
+            "SaveONNX": true,
+            "ONNXPrefix": "my_model",
+            "SaveONNXPath": "/path/to/onnx/directory"
+        }
+    }
+
+    Args:
+        ortmodule (:obj:`ORTModule`): ORTModule instance that needs to be configured
+        path (:obj:`str`, optional): Path to json file. Alternatively, users can set the
+            environement variable ORTMODULE_JSON_CONFIG_PATH to the json config path. In case
+            both path and environment variable are set, the environment variable gets precedence.
+    """
+
+    global JSON_PATH_ENVIRONMENT_KEY
+    JSON_PATH_ENVIRONMENT_KEY = "ORTMODULE_JSON_CONFIG_PATH"
+    path = os.getenv(JSON_PATH_ENVIRONMENT_KEY, path)
+
+    # figure out the json path
+    if path is None:
+        raise ValueError(f"Path to json is not provided. Provide the path through function call or by setting the environment variable {json_path_environment_key}")
+
+    # load the entire json file
+    data = _load_data_from_json(path)
```

Comment:
Are you going to create a follow-up PR to check whether a key in json is not part of any possible keys (defined at `_define_load_function_keys`)?

That will probably be helpful for users as the schema changes with time. They need to know what has been obsolete or what is a typo

They can define a `SaveOnnx` as opposed to `SaveONNX` and spend hours running CI just to learn the ONNX will not be saved and their debug time was wasted